dataRetriever/resolvers/headerResolver.go

```diff
@@ -113,18 +119,31 @@ func (hdrRes *HeaderResolver) ProcessReceivedMessage(message p2p.MessageP2P, _ f
 	return hdrRes.Send(buff, message.Peer())
 }
 
-func (hdrRes *HeaderResolver) resolveHeaderFromNonce(key []byte) ([]byte, error) {
+func (hdrRes *HeaderResolver) resolveHeaderFromNonce(rd *dataRetriever.RequestData) ([]byte, error) {
 	// key is now an encoded nonce (uint64)
 
 	// Search the nonce-key pair in cache-storage
-	hash, err := hdrRes.hdrNoncesStorage.Get(key)
+	hash, err := hdrRes.hdrNoncesStorage.GetFromEpoch(rd.Value, rd.Epoch)
 	if err != nil {
 		log.Trace("hdrNoncesStorage.Get", "error", err.Error())
+		nonce, err := hdrRes.nonceConverter.ToUint64(rd.Value)
```

Comment:
as I see this is not implemented. We should use SearchFirst in storage everywhere now.

Reply:
why would we need the epoch provider by nonce component then ?